### ISSUE TITLE
[@typescript-eslint/quotes] Backticks confuse compiler with types in switch

### ISSUE THREAD

**Issue by BBosman** [typescript-eslint/typescript-eslint#1059]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/quotes": ["error", "backtick", { "avoidEscape": true }]
  }
}
```

```TS
const value = myVar as unknown;
switch (typeof value) {
  case `object`:
    if (value === null) {
      valueText = `null`;
    } else {
      valueText = _ctorName(value);
    }
    break;
  case `number`:
    valueText = value.toString();
    break;
  // ...
  default:
    valueText = _ctorName(value as Instance);
}
```

**Expected Result**
That the Typescript compiler knows the type of `value` within the individual case statements.

**Actual Result**
The compiler still thinks `value` is of type `unknown`.

**Additional Info**
If I change the backticks in the `case` statements to single quotes it works, so I think the best way to solve this, is to leave those as single quotes and have the`@typescript-eslint/quotes` rule be okay with that.

It also works with backticks with `typescript@3.7.0-dev.20191009`, but that is not (yet) a supported version. So the above fix would only be required for `typescript@<3.7.0`.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.3.3` |
| `@typescript-eslint/parser`        | `2.3.3` |
| `TypeScript`                       | `3.6.3` |
| `ESLint`                           | `6.5.1` |
| `node`                             | `10.15.0` |
| `npm`                              | `6.11.3` |

**Comment by bradzacher** [comment]
Looking at a simplified example in the ts playground
```ts
const value: unknown = null;
switch (typeof value) {
  case "string":
    console.log(value.length);
    break;
  case 'number':
    console.log(value.valueOf());
    break;
  case `object`:
    console.log(value);
    break;
}
```
[repl](https://www.typescriptlang.org/play/?ssl=9&ssc=15&pln=9&pc=9#code/MYewdgzgLgBAbgQwDYFcCmAuGKwGswgDuYMAvDGCkkgNwBQEhAllMABYwAUUAngA5oQAM3jJ0AShgBvOjBjAEENDABE0AE5MwAcxUZZc+eAggkaAHRIQ2zolQWzOqG3H1DMAEbq0CXG-mKygDklAC2HmjqQfruoJCmDta2YhZ26ADyQpzirgZyXj5+BgpKMAAGIB4AVmjAUGUxhnEmZpZJaWi57gW+9AC+QA)

It looks like this is just a bug in TS3.6
With the the backtick case there, it causes typescript to not handle the type narrowing, at all!
However if you remove the backtick case, then the other two cases start working.

I'm not sure if we should be working around a bug in TS? Additionally - using `backtick` as a value is a very rare case.
Happy to accept a PR if you think there's value in adding this case.

**Comment by BBosman** [comment]
I'll just downgrade them to warnings for now and switch back to errors when TS3.7 is released.

It's indeed not worth the cycles for something that will fix itself automatically soonish.

Thanks for looking into it.
